superset/assets/javascripts/SqlLab/actions.js

```diff
@@ -382,3 +386,38 @@ export function popSavedQuery(saveQueryId) {
     });
   };
 }
+
+export function createDatasourceStarted() {
+  return { type: CREATE_DATASOURCE_STARTED };
+}
+export function createDatasourceSuccess(response) {
+  const data = JSON.parse(response);
+  const datasource = `${data.table_id}__table`;
+  return { type: CREATE_DATASOURCE_SUCCESS, datasource };
+}
+export function createDatasourceFailed(err) {
+  return { type: CREATE_DATASOURCE_FAILED, err };
+}
+
+export function createDatasource(vizOptions, context) {
+  return (dispatch) => {
+    dispatch(createDatasourceStarted());
+
+    return $.ajax({
+      type: 'POST',
+      url: '/superset/sqllab_viz/',
+      async: false,
```

Comment:
it was in the original code:
https://github.com/airbnb/superset/blob/ac3aba7c7da905f0f4bfeb0f80efb87136812b90/superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

But i also think it is not necessary...Will remove it.